tests/testthat/test-helpers.R

```diff
@@ -125,4 +125,24 @@ test_that("model can be saved and loaded", {
 
   expect_type(out, "double")
   expect_length(out, 51)
+
+  # Remove record_evals from the saved model zipfile to make sure the model can
+  # still be loaded if record_evals is missing
+  ex_dir <- glue::glue("{tempdir()}/no_record_evals")
+  zip::unzip(file, exdir = ex_dir)
+  file.remove(
+    c(
+      glue::glue("{ex_dir}/record_evals.model"),
+      # We have to remove record_evals.model from the tempdir as well, or else
+      # it will still exist when lgbm_load uses the tempdir to extract
+      # the new zipfile
+      glue::glue("{tempdir()}/record_evals.model")
+    )
+  )
+  new_file <- tempfile(fileext = ".zip")
+  zip::zipr(new_file, files = list.files(ex_dir, full.names = TRUE))
```

Comment:
This is a little bit hacky, but it was the easiest way I could think of to test the change.